### PR TITLE
Resolves #313

### LINKED ISSUE #313: Improve Javadoc of API methods with regards to `access` vs `dependOn`
A follow up of #302, i.e. the Javadoc on `ClassesShould#accessClassesThat()` and similar (all methods dealing with `access` and `dependsOn`) should be improved.

In particular methods `access...` should explain the difference and refer to `dependOn` and vice versa.

### PR DESCRIPTION
**Pull request by idosal**
Resolves: #313 (Improve Javadoc of API methods with regards to 'access' vs 'dependOn').

### CODE REVIEW

**Comment by codecholeric** [comment]
Thanks a lot for improving this :+1: :smiley: I think this will take away some confusion users always have about the difference of these methods.

**Review by codecholeric — APPROVED** [review]

File: archunit/src/main/java/com/tngtech/archunit/lang/syntax/elements/ClassesShould.java [hunk 1]
```diff
@@ -821,6 +821,9 @@
      * {@link ArchRuleDefinition#noClasses() noClasses()}.{@link GivenClasses#should() should()}.{@link #accessClassesThat()}.{@link ClassesThat#haveFullyQualifiedName(String) haveFullyQualifiedName(String)}
      * </code></pre>
      *
+     * NOTE: 'access' refers only to violations by real accesses, i.e. accessing a field, and calling a method.
+     * Compare with {@link #dependOnClassesThat()} (DescribedPredicate)} that catches a wider variety of violations.
```
Comment (hankem) [inline archunit/src/main/java/com/tngtech/archunit/lang/syntax/elements/ClassesShould.java:825]: Should the `(DescribedPredicate)}` be here?

Comment (codecholeric) [reply archunit/src/main/java/com/tngtech/archunit/lang/syntax/elements/ClassesShould.java:825]: Oh true, good spot, I don't think so :joy:

Comment (codecholeric) [reply archunit/src/main/java/com/tngtech/archunit/lang/syntax/elements/ClassesShould.java:825]: That looks like the typical autocorrect IntelliJ always gets me with :wink:

Comment (codecholeric) [reply archunit/src/main/java/com/tngtech/archunit/lang/syntax/elements/ClassesShould.java]: I took the liberty to fix that up :+1:

**Pull request merged**